presto-main/src/main/java/com/facebook/presto/execution/scheduler/SectionExecutionFactory.java

```diff
@@ -335,8 +335,8 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     // verify execution is consistent with planner's decision on dynamic lifespan schedule
                     verify(bucketNodeMap.isDynamic() == dynamicLifespanSchedule);
 
-                    if (!bucketNodeMap.isDynamic()) {
-                        stageNodeList = ((FixedBucketNodeMap) bucketNodeMap).getBucketToNode().stream()
+                    if (!bucketNodeMap.isDynamic() || bucketNodeMap.getBucketToNode().isPresent()) {
```

Comment:
In this case `!bucketNodeMap.isDynamic()` seems a bit redundant?

Reply:
> In this case `!bucketNodeMap.isDynamic()` seems a bit redundant?

if we change `bucketNodeMap.getBucketToNode().isPresent()` to other function, we may need to keep it